src/lib/yearn-affiliate-wrapper.sol

```diff
@@ -0,0 +1,357 @@
+
+    function maxAvailableShares() external view returns (uint256);
+
+    /**
+     * View how much the Vault would increase this Strategy's borrow limit,
+     * based on its present performance (since its last report). Can be used to
+     * determine expectedReturn in your Strategy.
+     */
+    function creditAvailable() external view returns (uint256);
+
+    /**
+     * View how much the Vault would like to pull back from the Strategy,
+     * based on its present performance (since its last report). Can be used to
+     * determine expectedReturn in your Strategy.
+     */
+    function debtOutstanding() external view returns (uint256);
+
+    /**
+     * View how much the Vault expect this Strategy to return at the current
+     * block, based on its present performance (since its last report). Can be
+     * used to determine expectedReturn in your Strategy.
+     */
+    function expectedReturn() external view returns (uint256);
+
+    /**
+     * This is the main contact point where the Strategy interacts with the
+     * Vault. It is critical that this call is handled as intended by the
+     * Strategy. Therefore, this function will be called by BaseStrategy to
+     * make sure the integration is correct.
+     */
+    function report(
+        uint256 _gain,
+        uint256 _loss,
+        uint256 _debtPayment
+    ) external returns (uint256);
+
+    /**
+     * This function should only be used in the scenario where the Strategy is
+     * being retired but no migration of the positions are possible, or in the
+     * extreme scenario that the Strategy needs to be put into "Emergency Exit"
+     * mode in order for it to exit as quickly as possible. The latter scenario
+     * could be for any reason that is considered "critical" that the Strategy
+     * exits its position as fast as possible, such as a sudden change in
+     * market conditions leading to losses, or an imminent failure in an
+     * external dependency.
+     */
+    function revokeStrategy() external;
+
+    /**
+     * View the governance address of the Vault to assert privileged functions
+     * can only be called by governance. The Strategy serves the Vault, so it
+     * is subject to governance defined by the Vault.
+     */
+    function governance() external view returns (address);
+
+    /**
+     * View the management address of the Vault to assert privileged functions
+     * can only be called by management. The Strategy serves the Vault, so it
+     * is subject to management defined by the Vault.
+     */
+    function management() external view returns (address);
+
+    /**
+     * View the guardian address of the Vault to assert privileged functions
+     * can only be called by guardian. The Strategy serves the Vault, so it
+     * is subject to guardian defined by the Vault.
+     */
+    function guardian() external view returns (address);
+}
+
+abstract contract YearnAffiliateWrapper {
```

Comment:
Could be good to update to latest BaseWrapper to get latest changes in sync

https://github.com/yearn/yearn-vaults/blob/master/contracts/BaseWrapper.sol